xla/translate/mhlo_to_hlo/mlir_hlo_to_hlo.cc

```diff
@@ -1095,8 +1123,8 @@ LogicalResult ExportXlaOp(AsyncStartOp op, OpLoweringContext ctx) {
       dyn_cast_or_null<AllGatherOp>(callee.getBody().front().front());
   if (all_gather_op && SimplyReturnedOp(all_gather_op)) {
     TensorType operand_type =
-        all_gather_op.getOperand().getType().cast<TensorType>();
-    TensorType result_type = all_gather_op.getType();
+        all_gather_op.getOperand(0).getType().cast<TensorType>();
+    TensorType result_type = all_gather_op.getType(0).cast<TensorType>();
     if (!operand_type.hasStaticShape() || !result_type.hasStaticShape())
       return failure();
     if (operands.size() != 1) return failure();
```

Comment:
Let's raise this check before the call to `getOperand(0)`

Reply:
`mhlo/IR/hlo_ops.td` defines that `MHLO_AllGatherOp` has `Variadic<MHLO_Tensor>:$operands,`
I tried to use `mhlo.all_gather()` without operands in export.mlir - got error
```
error: unexpected error: 'mhlo.all_gather' op expected 1 or more operands, but found 0
  %0:2 = "mhlo.all_gather"() {
```
mhlo parser validates that MHLO_AllGatherOp has at least one operand.
It should be safe to use `all_gather_op.getOperand(0)` in `mlir_hlo_to_hlo.cc::ExportXlaOp` since mlir validation happens before ExportXlaOp call.